sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -2309,8 +2404,29 @@ CurlConnection::CurlConnection(
         _detail::DefaultFailedToGetNewConnectionTemplate + hostDisplayName + ". "
         + std::string("curl_easy_init returned Null"));
   }
+
   CURLcode result;
 
+  if (request.GetHeader("connection").HasValue() && request.GetHeader("keep-alive").HasValue())
+  {
+    auto connectionHeader = Azure::Core::_internal::StringExtensions::ToLower(
+        request.GetHeader("connection").Value());
```

Comment:
Because I had to look this up three times, it might be helpful to add a comment indicating that while the header maps are case insensitive, you need to lower case the *value* of the entry in the map.